platform/log/zephyr/src/mender-log.c

```diff
@@ -33,36 +33,47 @@ mender_log_init(void) {
     return MENDER_OK;
 }
 
+/* IMPORTANT: Please note that the default size of internal Zephyr logger buffer is 1024 bytes; if we need log buffer size 
+              to be more than that there is CONFIG_LOG_BUFFER_SIZE option allowing to extend the buffer size. */
+#define LOG_BUFFER_SIZE_BYTES (128)
+
 mender_err_t
 mender_log_print(uint8_t level, const char *filename, const char *function, int line, char *format, ...) {
 
+    (void)filename;
     (void)function;
-    char log[256] = { 0 };
 
-    /* Format message */
     va_list args;
     va_start(args, format);
-    vsnprintf(log, sizeof(log), format, args);
-    va_end(args);
+
+    char log_buff[LOG_BUFFER_SIZE_BYTES];
+    int  ret = vsnprintf(log_buff, LOG_BUFFER_SIZE_BYTES, format, args);
+
+    if (ret < 0) {
+        LOG_ERR("logger error: log message formatting failed: %d", ret);
+        va_end(args);
+        return MENDER_FAIL;
+    }
```

Comment:
Could we check `ret > LOG_BUFFER_SIZE_BYTES` and replace the end of the buffer with `(truncated)` or `...`?

Reply:
Yes, but should be `>=`, because it returns number of characters that would have been printed, not including the trailing `'\0'`.